reference/language-concepts/namespaces.markdown

```diff
@@ -6,104 +6,76 @@ sorting: 100
-[file control promise][file control#namespace] to the file before the relevant
-bundles or bodies. All bundles and bodies start off in the `default` namespace
-if you don't explicitly set this. Once set, this applies until the end of the
-file or the next namespace change.
+Namespaces are declared with [`body file control`][file control#namespace]. A
+namespace applies within a single file to all subsequently defined bodies and bundles
+following the namespace declaration until a different namespace has been
+declared or until the end of the file.
 
-```cf3
-    body file control
-    {
-       namespace => "myspace";
-    }
-```
+[%CFEngine_include_example(namespace_declaration.cf)%]
 
-### Accessing syntax elements between namespaces and the default namespace
+**Notes:**
 
-To distinguish the bundle `mymethod` in the default namespace from one in
-another namespace, you prefix the bundle name with the namespace, separated by
-a colon.
+- Multiple namespaces can be declared within the same file
+- The same namespace can be declared in multiple files
+- The same namespace can be declared in the same file multiple times
 
-```cf3
-    methods:
+## Methods|usebundle
 
-      "namespace demo" usebundle => myspace:mymethod("arg1");
-      "namespace demo" usebundle => mymethod("arg1","arg2");
-```
+Methods promises assume you are referring to a bundle in the same namespace as
+the promiser. To refer to a bundle in another namespace you *must* specify the
+namespace by prefixing the bundle name with the namespace followed by a colon
+(`:`).
 
-To distinguish a body from one in another namespace, you can prefix the body name with the namespace, separated by a colon.
+[%CFEngine_include_example(namespace_methods-usebundle.cf)%]
 
-```cf3
-    files:
-       "/file"
-          create => "true",
-           perms => name1:settings;
-```
+## Bodies
 
-If you don't make any namespace declarations, you'll be in the
-`default` namespace.  Bundles, bodies, classes, and variables from the
-`default` namespace can be accessed like any other:
+Bodies are assumed to be within the same namespace as the promiser. To use a body from another namespace the namespace must be specified by prefixing the body name with the namespace followed by a colon (`:`).
 
-```cf3
-    files:
-      "/file"
-         create => "true",
-          perms => default:settings;
-```
+A common mistake is forgetting to specify `default:` when using bodies from the standard library which resides in the `default` namespace.
 
-If you use the standard library from your own namespace, remember to
-specify this `default:` prefix.
+[%CFEngine_include_example(namespace_bodies.cf)%]
 
-To access classes, variables, or meta-data in bundles in a different namespace, use the
-colon as a namespace prefix:
+## Variables
 
-    $(namespace:bundle.variable)
-    $(namespace:bundle_meta.variable)
+Variables (except for Special Variables) are assumed to be within the same scope
```

Comment:
s/scope/namespace/ , the term scope is not used any where else in this doc so using "namespace" will be more clear.

Reply:
Yes, good suggestion.

Reply:
Reading this again .... I said scoped because i was also referring to how `$(variable)` is == to `$($(this.bundle).variable)` which is == to `$($(this.namespace):$(this.bundle).variable)`

Which I was trying to illustrate in referenced core example: https://github.com/cfengine/core/pull/4966/files#diff-5e851531f57a16834042323a690dbbf9db12336174725839130109d23fa519f7